src/main/java/com/onlinebookstore/onlinebookstore/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package com.onlinebookstore.onlinebookstore.service.impl;
+
+import com.onlinebookstore.onlinebookstore.dto.user.UserRegistrationRequestDto;
+import com.onlinebookstore.onlinebookstore.dto.user.UserResponseDto;
+import com.onlinebookstore.onlinebookstore.exeption.RegistrationException;
+import com.onlinebookstore.onlinebookstore.mapper.UserMapper;
+import com.onlinebookstore.onlinebookstore.model.User;
+import com.onlinebookstore.onlinebookstore.repository.UserRepository;
+import com.onlinebookstore.onlinebookstore.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+
+@Service
+@RequiredArgsConstructor
+@Validated
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {

```

Comment:
a bit more efficient and simple
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```